eip-5564.md

```diff
@@ -99,23 +99,33 @@ This specification additionally defines a singleton `ERC5564Messenger` contract
 contract IERC5564Messenger {
   /// @dev Emitted when sending something to a stealth address.
   /// @dev See the `announce` method for documentation on the parameters.
-  event Announcement(address indexed stealthAddress, bytes ephemeralPubKey, bytes32 metadata);
+  event Announcement (
+    uint256 indexed schemeID, 
+    address indexed stealthAddress, 
+    bytes ephemeralPubKey, 
+    bytes metadata
+  );
 
   /// @dev Called by integrators to emit an `Announcement` event.
+  /// @param schemeID The applied stealth address scheme (f.e. secp25k1).
   /// @param stealthAddress The computed stealth address for the recipient.
   /// @param ephemeralPubKey Ephemeral public key used by the sender.
   /// @param metadata An arbitrary field that the sender can use however they like, but the below
   /// guidelines are recommended:
-  ///   - The first byte of the metadata MUST be the stealth address scheme ID.
-  ///   - The second byte of the metadata MUST be the view tag.
+  ///   - The first byte of the metadata MUST be the view tag.
   ///   - When sending ERC-20 tokens, the metadata SHOULD include the token address as the next 20
-  ///     bytes, and the amount being sent in the remaining 10 bytes.
+  ///     bytes, and the amount being sent in the remaining 11 bytes.
   ///   - When sending ERC-721 tokens, the metadata SHOULD include the token address as the first 20
-  ///     bytes, and the token ID being sent as remaining 10 bytes.
-  function announce(address stealthAddress, bytes memory ephemeralPubKey, bytes32 metadata)
+  ///     bytes, and the token ID being sent as remaining 11 bytes.
```

Comment:
Now that we have more space, what do you think of this instead?

- The first byte of the metadata MUST be the view tag.
- When sending ERC-20 tokens, the metadata SHOULD be structured as follows:
    - Byte 1 MUST be the view tag, as specified above.
    - Bytes 2-5 are `23b872dd`, which the signature for the ERC-20 `transferFrom` method.
    - Bytes 6-25 are the token address.
    - Bytes 26-57 are the amount of tokens being sent.
- When approving a stealth address to spend ERC-20 tokens, the metadata SHOULD be structured as follows:
    - Byte 1 MUST be the view tag, as specified above.
    - Bytes 2-5 are `0xe1f21c67`, which the signature for the ERC-20 `approve` method.
    - Bytes 6-25 are the token address.
    - Bytes 26-57 are the approval amount.
- When sending ERC-721 tokens, the metadata SHOULD be structured as follows:
    - Byte 1 MUST be the view tag, as specified above.
    - Bytes 2-5 are `23b872dd`, which the signature for the ERC-721 `transfer` method.
    - Bytes 6-25 are the token address.
    - Bytes 26-57 are the token ID of the token being sent.

Reply:
I think that 2-5 bytes it an overkill for the method id because it's already a the hashed canonical function and therefore quite unique. I think that 1 byte is more than enough, 2 at max because realistically, it's either a standard eth transaction or a transfer in 99% of the time and if it's an approval or another interaction, 1 byte should be sill enough to distinguish that.
On the other hand, what benefits do we provide an application if it knows, without looking into the contract or performing rpc calls, which exact function that was executed in a stealth interaction. F.e. if it's an approval, then the announcement event would already provide enough info for the app to suggest to execute an TransferFrom call.

In general, I really like the idea and think it makes the announcement much more valuable, though, we're not saving gas at all. Do you think we can use that much space, even though it is never stored and just temorary info emitted in an event?

Reply:
> I think that 2-5 bytes it an overkill for the method id because it's already a the hashed canonical function and therefore quite unique. I think that 1 byte is more than enough, 2 at max because realistically, it's either a standard eth transaction or a transfer in 99% of the time and if it's an approval or another interaction, 1 byte should be sill enough to distinguish that.

I agree with all of this, in theory 1-2 bytes should be fine.

However, my thinking was that since we're trying to future-proof this standard (e.g. by using `bytes memory` for private keys instead of `uint256`), it seems we should be consistent and future proof here too—you never know what future ERCs/new (token) standards might emerge and have a 1-2 byte clash with approve or transferFrom. The cost of emitting a few extra bytes is quite small, so it seems like an ok tradeoff to me